### PR TITLE
Feat/pupil 212/owa short answer

### PR DESCRIPTION
**Pull request by kimon-satan**
### User Story

Attempt short answer questions and get feedback

### Implementation details

Put as much information here as possible about how to do this task

refactor:

- Move the useffect into callback with a switch statement in QuizRenderer
- Create an onInitialChange callback and do the same

### Acceptance criteria

- [x]  refactor mcqs to remove useEffect (As per above)
- [x]  short answer rendering
- [x]  short answer grading
- [x]  test coverage
- [x]  documentation where relevant

### to test

https://deploy-preview-2178--oak-web-application.netlify.thenational.academy/pupils/programmes/english-primary-ks2/units/crazy-about-cats-reading/lessons/analysing-use-of-language-in-crazy-about-cats#starter-quiz

https://deploy-preview-2178--owa-storybook.netlify.thenational.academy/?path=/docs/components-pupiljourneycomponents-quizshortanswer--docs

### CODE REVIEW

File: src/components/PupilJourneyComponents/QuizShortAnswer/QuizShortAnswer.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    },
+  },
+} satisfies Meta<typeof QuizShortAnswer>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+// mock the QuizEngineContext
+const getContext = (
```
Comment (carlmw) [inline src/components/PupilJourneyComponents/QuizShortAnswer/QuizShortAnswer.stories.tsx:40]: I think i've been spoilt by libraries that auto mock through introspection but this looks annoying to have written 😆

File: src/components/PupilJourneyComponents/QuizShortAnswer/QuizShortAnswer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+import {
+  QuizEngineContextType,
+  QuizEngineContext,
+} from "@/components/PupilJourneyComponents/QuizEngineProvider";
+import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";
+import { quizQuestions } from "@/node-lib/curriculum-api-2023/fixtures/quizElements.fixture";
+
+const shortAnswerQuestion = quizQuestions?.find(
+  (q) => q.answers?.["short-answer"] && q.answers?.["short-answer"].length > 0,
```
Comment (carlmw) [inline src/components/PupilJourneyComponents/QuizShortAnswer/QuizShortAnswer.test.tsx:17]: Another potential use for those discriminators I mentioned! `isShortAnswer` 

```typescript
const shortAnswerQuestion = quizQuestions?.find((q) => isShortAnswer(q.answers));
```

**Review by carlmw — APPROVED** [review]
`useEffect` must be destroyed at all costs 🙇🏼 

killitwithfire

Otherwise no blockers from me, just thoughts for future refactors.

**Comment by rightsaidjames** [comment]
There is no visible label on the input field, as per the designs (this is a fill in the blank question, but the question type is the same in the data):
Screenshot 2024-01-11 at 09 29 58

But we need further discussion on this issue, as well as an equivalent solution for MCQ question types, so I'm happy to approve this ticket as-is and tackle the problem in follow-up ticket(s).

Looks good, otherwise!

**Review by rightsaidjames — APPROVED** [review]

File: src/components/PupilJourneyComponents/QuizShortAnswer/QuizShortAnswer.test.tsx (hunk 2 above)
Comment (kimon-satan) [reply src/components/PupilJourneyComponents/QuizShortAnswer/QuizShortAnswer.test.tsx:17]: Okay. You've convinced me.  Lets work out a ticket where we can slot this stuff in.

**Pull request merged**